cmd/expired-authz-purger/main.go

```diff
@@ -71,33 +91,30 @@ func (p *expiredAuthzPurger) purgeAuthzs(purgeBefore time.Time, yes bool) (int64
 		}
 	}
 
-	rowsAffected := int64(0)
-	for {
-		result, err := p.db.Exec(`
-			DELETE FROM pendingAuthorizations
-			WHERE expires <= ?
-			LIMIT ?
-			`,
-			purgeBefore,
-			p.batchSize,
-		)
+	for _, id := range ids {
+		// Delete challenges + authorization
+		_, err := p.db.Exec("DELETE FROM challenges WHERE authorizationID = ?", id)
```

Comment:
Can you add a comment here noting that we only delete the authz if the challenge deletion was successful, so we are safe from stranding challenges?